toolchain/parse/handle_var.cpp

```diff
@@ -47,7 +47,13 @@ auto HandleVarAsReturned(Context& context) -> void {
 }
 
 auto HandleVarAsFor(Context& context) -> void {
-  HandleVar(context, State::VarFinishAsFor);
+  auto state = context.PopState();
+
+  // The finished variable declaration will start at the `var`.
+  state.state = State::VarFinishAsFor;
+  context.PushState(state);
```

Comment:
Looks like we have a simpler way to write this now:
```suggestion
  context.PushState(state, State::VarFinishAsFor);
```